docs/README-node-utils-APIs-xr.md

```diff
@@ -0,0 +1,277 @@
+# How To Update node_utils APIs to Support XR
+
+#### Table of Contents
+
+* [Overview](#overview)
+* [Run minitest on Nexus](#minitest-nexus)
+* [Run minitest on XR](#minitest-xr)
+* [Fix errors](#fix-errors)
+ * [get_command differences](#get_command)
+ * [get_value/set_value differences](#get_value)
+ * [Unsupported attributes](#unsupported)
+ * [Interface differences](#interface)
+ * [Setup/teardown differences](#setup)
+ * [Dependency differences](#dependency)
+ * [Other differences](#other)
+
+## <a name="overview">Overview</a>
+
+This document is a HowTo guide for taking existing cisco_node_utils APIs written to work on Nexus, and enhancing them to support IOS XR. Please see [How To Create New node_utils APIs](./README-develop-node-utils-APIs.md) for setup steps.
+
+Specifically, this will walk you through the general steps that were needed to enhance the bgp_neighbor feature to add XR support.  The changes detailed here are not the only changes that were made, but they should be representative of the kinds of changes that might be needed.
+
+*Note: This document assumes you have a Nexus VM/device at 192.168.0.1 with user/password admin/admin, and an XR VM/device at 192.168.0.2 with user/password admin/admin and grpc port configured to 57777.*
+
+## <a name="minitest-nexus">Run minitest on Nexus</a>
+
+It's a good idea to make sure all the tests are running on Nexus before making any changes (just to be sure).  From the ./cisco_network_node_utils/tests directory:
+
+```bash
+% ruby test_bgp_neighbor.rb -v -- 192.168.0.1 admin admin
+```
+
+All tests should finish successfully (no failures, errors, or skips).
+
+## <a name="minitest-xr">Run minitest on XR</a>
+
+Now run the same tests against XR, to see what is broken and to get an idea of how much work will need to be done.
+
+```bash
+% ruby test_bgp_neighbor.rb -v -- 192.168.0.2:57777 admin admin
+```
+
+## <a name="fix-errors">Fix errors</a>
+
+We'll now go through the types of differences that will cause errors/failures, and how to fix them.  When you've fixed a particular error, run minitest again on that single test until it is working as you intend, then re-run the same test against Nexus to make sure it still works.
+
+#### <a name="get_command">get_command differences</a>
+
+The base get_command parameter will often be different on XR than Nexus.  Usually, the get_command is common across many attributes of a given feature, and therefore will be defined in the _template section.  In the case of the bgp neighbor feature, the get_command parameter to retrieve information about neighbors is "show running bgp all" on Nexus, while it is "show running router bgp" on XR.  This difference caused most/all of the test_bgp_neighbor.rb tests to error when run against XR, which looked something like this:
```

Comment:
General nitpick - please format literal strings, filenames, and YAML entries in `monospace`. This would include `get_command`, `_template`, `"show running bgp all"`, `test_bgp_neighbor.rb`. Thanks.